lapis/job_io/htcondor.py

```diff
@@ -31,36 +32,55 @@ def htcondor_job_reader(
         "DiskUsage_RAW": 1.024 / 1024 / 1024,
     },
 ):
-    htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
-
-    for row in htcondor_reader:
-        if float(row[used_resource_name_mapping["walltime"]]) <= 0:
+    input_file_type = iterable.name.split(".")[-1]
+    if input_file_type == "json":
+        htcondor_reader = json.load(iterable)
+    elif input_file_type == "csv":
+        htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    else:
+        logging.getLogger("implementation").error(
+            "Invalid input file %s. Job input file can not be read." % iterable.name
+        )
+    for entry in htcondor_reader:
+        if float(entry[used_resource_name_mapping["walltime"]]) <= 0:
             logging.getLogger("implementation").warning(
-                "removed job from htcondor import (%s)", row
+                "removed job from htcondor import (%s)", entry
             )
             continue
         resources = {}
         for key, original_key in resource_name_mapping.items():
             try:
-                resources[key] = float(row[original_key]) * unit_conversion_mapping.get(
-                    original_key, 1
-                )
+                resources[key] = float(
+                    entry[original_key]
+                ) * unit_conversion_mapping.get(original_key, 1)
             except ValueError:
                 pass
+
         used_resources = {
             "cores": (
-                (float(row["RemoteSysCpu"]) + float(row["RemoteUserCpu"]))
-                / float(row[used_resource_name_mapping["walltime"]])
+                (float(entry["RemoteSysCpu"]) + float(entry["RemoteUserCpu"]))
+                / float(entry[used_resource_name_mapping["walltime"]])
             )
             * unit_conversion_mapping.get(used_resource_name_mapping["cores"], 1)
         }
         for key in ["memory", "walltime", "disk"]:
             original_key = used_resource_name_mapping[key]
             used_resources[key] = float(
-                row[original_key]
+                entry[original_key]
             ) * unit_conversion_mapping.get(original_key, 1)
+
+        try:
+            resources["inputfiles"] = {
+                file["filename"]: {
+                    "filesize": file["filesize"],
+                    "usedsize": file["usedsize"],
+                }
+                for file in entry["Inputfiles"]
+            }
```

Comment:
I'm changing the structure of the dictionary. The relevant part of the input file is 

> 		"Inputfiles": [
> 			{
> 				"filename": "a.root",
> 				"filesize": 25000,
> 				"usedsize": 20000
> 			},
> 			{
> 				"filename": "b.root",
> 				"filesize": 25000,
> 				"usedsize": 20000
> 			},
> 			{
> 				"filename": "c.root",
> 				"filesize": 25000,
> 				"usedsize": 20000
> 			}
> 		]
The dictionary that I'm creating in the job file readout has a different structure. I'm using the "filename" entry as dictionary key and "filesize"/"usedsize" as nested dictionary. 
Would you prefer a more compact way to create this dictionary like 
```
file["filename"]: {key: file[key] for key in file.keys() if key != "filename}
                for file in entry["Inputfiles"]
            
```
?

Reply:
My question is not related to the compactness of the code. No matter what, code should be readable. I was only wondering about the JSON input file you defined. The specification you propose, does it follow any known format or did you decide on this? What is your reasoning on having this flat structure?

Reply:
I don't see how I could use the nested dictionary structure in the json file because I need a key ("filename", "filesize",...) for every input file property and using the filename as key would destroy the read out (`entry["filename"]`) because I wouldn't have the "filename"-key.

Reply:
Or do you have an idea how the structure in the input file could be the same as in the nested dictionary? I'm new to json so maybe I just haven't seen it before...

Reply:
You might do the following structure in JSON:

```json
"Inputfiles": {
    "<filename>": {
        "requested": <filesize>,
        "used": <used_filesize>
    },
    ...
}
```

By the way, this formatting also prevents you from having duplicate file names :)

And in python you could easily do the following:

```python
resources["inputfiles"] = entry["Inputfiles"]
```

Reply:
I'll look into it, thanks :)